### PR TITLE
simplify stream association init

### PR DESCRIPTION
**Pull request by Kriechi**
Explicitly setting `stream_id = 0` for frames without a stream association is a bit verbose.
This PR removes this by defaulting the `stream_id` to `0`. 
Since the mixin classes require a `stream_id` argument for `init`, I had to keep it. Is there a cleaner solution?

### CODE REVIEW

File: hyperframe/frame.py [hunk 1]
```diff
@@ -30,7 +30,7 @@ class Frame(object):
     # it must be zero. If 'either', it's not checked.
     stream_association = None
 
-    def __init__(self, stream_id):
+    def __init__(self, stream_id=0):
```
Comment (Lukasa) [inline hyperframe/frame.py]: I think doing this for the base class runs the risk of hiding bugs in implementations of new frame types. I'm happy for the leaf frame classes to do this, I'm less happy about the base class doing it.

**Comment by Lukasa** [comment]
For the mixins, it's reasonable to adjust their constructors to take `*args` and `**kwargs` if that's what you're thinking.

File: hyperframe/frame.py (hunk 1 above)
Comment (Kriechi) [inline hyperframe/frame.py]: fixed.

**Comment by Kriechi** [comment]
I changed it to make use of `*args, **kwargs`. 
But because of [this](https://github.com/python-hyper/hyperframe/blob/2e28024273377b756725fcf7dee94a4dc714f01b/hyperframe/frame.py#L65) I had to leave the `stream_id` argument in the ctors for the frame classes without stream association.

**Comment by Lukasa** [comment]
Alright, this looks great to me. Thanks so much! :cake: :sparkles: :cake:

**Pull request merged**